apps/dep-graph-client/src/app/graph.ts

```diff
@@ -0,0 +1,384 @@
+import { default as tippy, Instance } from 'tippy.js';
+
+import cytoscapeDagre from 'cytoscape-dagre';
+import * as cy from 'cytoscape';
+import popper from 'cytoscape-popper';
+import anywherePanning from 'cytoscape-anywhere-panning';
+
+interface CytoscapeDagreConfig extends cy.BaseLayoutOptions {
+  // dagre algo options, uses default value on undefined
+  nodeSep: number; // the separation between adjacent nodes in the same rank
+  edgeSep: number; // the separation between adjacent edges in the same rank
+  rankSep: number; // the separation between each rank in the layout
+  rankDir: 'TB' | 'LR'; // 'TB' for top to bottom flow, 'LR' for left to right,
+  ranker: 'network-simplex' | 'tight-tree' | 'longest-path'; // Type of algorithm to assign a rank to each node in the input graph. Possible values: 'network-simplex', 'tight-tree' or 'longest-path'
+  minLen: (edge: cy.EdgeSingular) => number; // number of ranks to keep between the source and target of the edge
+  edgeWeight: (edge: cy.EdgeSingular) => number; // higher weight edges are generally made shorter and straighter than lower weight edges
+
+  // general layout options
+  fit: boolean; // whether to fit to viewport
+  padding: number; // fit padding
+  spacingFactor: number; // Applies a multiplicative factor (>0) to expand or compress the overall area that the nodes take up
+  nodeDimensionsIncludeLabels: boolean; // whether labels should be included in determining the space used by a node
+  animate: boolean; // whether to transition the node positions
+  animateFilter: (node, i) => boolean; // whether to animate specific nodes when animation is on; non-animated nodes immediately go to their final positions
+  animationDuration: number; // duration of animation in ms if enabled
+  animationEasing: string; // easing of animation if enabled
+  boundingBox:
+    | { x1: number; y1: number; x2: number; y2: number }
+    | { x1: number; y1: number; w: number; h: number }; // constrain layout bounds; { x1, y1, x2, y2 } or { x1, y1, w, h }
+  transform: (node, pos) => cy.Position; // a function that applies a transform to the final node position
+  ready: () => void; // on layoutready
+  stop: () => void; // on layoutstop
+}
+
+enum NrwlPallette {
+  blue = '#48c4e5',
+  lightBlue = '#96d8e9',
+  gray = '#333333',
+  navy = '#143055',
+  twilight = '#086c9f',
+  black = '#231f20',
+  red = 'rgb(248, 84, 119)',
+  white = '#fff',
+}
+const FONTS = '"Helvetica Neue", sans-serif';
+
+export class GraphComponent {
+  private graph: cy.Core;
+  private openTooltip: Instance = null;
+
+  constructor() {
+    cy.use(cytoscapeDagre);
+    cy.use(popper);
+    cy.use(anywherePanning);
+  }
+
+  createCytoScapeDirectoryParents(
+    elements: cy.ElementDefinition[],
+    directories: string[]
+  ): cy.ElementDefinition[] {
+    let childDirectory = directories.join('/');
```

Comment:
I guess maybe we should also separate out the logic to create a directory of projects?